drivers/flash/soc_flash_nrf.c

```diff
@@ -23,14 +23,29 @@
 #include "controller/ticker/ticker.h"
 #include "controller/include/ll.h"
 
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+
+/* The timeout is multiplied by 1.5 because switching tasks may take
+ * significant portion of time.
+ */
+#define FLASH_TIMEOUT_MS ((FLASH_PAGE_ERASE_MAX_TIME_US)\
+			* (FLASH_PAGE_MAX_CNT) / 1000 * 15 / 10)
+
+#else
+
+#define FLASH_TIMEOUT_MS ((FLASH_PAGE_ERASE_MAX_TIME_US)\
+			* (FLASH_PAGE_MAX_CNT) / 1000)
+
+#endif /* CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE */
+
 #define FLASH_SLOT_ERASE     FLASH_PAGE_ERASE_MAX_TIME_US
```

Comment:
Should FLASH_SLOT_ERASE be now a smaller value?

Ticker reservatiosn should be in smaller units if flash erase is partial and does not take ~80-90 ms on nRF52840.